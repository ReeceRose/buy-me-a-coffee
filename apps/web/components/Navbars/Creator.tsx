import { useWallet } from '@solana/wallet-adapter-react';
import {
  WalletDisconnectButton,
  WalletMultiButton,
} from '@solana/wallet-adapter-react-ui';
import Link from 'next/link';

import { BiHome } from 'react-icons/bi';

export const CreatorNavbar = (): JSX.Element => {
  const { connected } = useWallet();

  return (
    <nav className="fixed top-0 z-50 flex flex-wrap items-center justify-between w-full px-2 py-3 bg-transparent">
      <div className="container flex flex-wrap items-center justify-between px-4 mx-auto">
        <div className="flex items-center flex-grow shadow-none " id="nav">
          <ul className="flex flex-row ml-auto list-none">
            <li className="flex items-center">
              <Link href="/">
                <a className="mr-2 text-lg text-white leading-lg">
                  <span className="flex items-center">
                    <BiHome className="mr-2" />
                    Home
                  </span>
                </a>
              </Link>
            </li>
            <li className="flex items-center">
              {connected ? <WalletDisconnectButton /> : <WalletMultiButton />}
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
