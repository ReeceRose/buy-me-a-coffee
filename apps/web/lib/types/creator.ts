export type Creator = {
  username: string;
  description: string;
  bannerImageUrl: string;
  profileImageUrl: string;
  supporters: number;
  support: Support;
  wallet: string;
};

export type Support = {
  icon: string;
  item: string;
  defaults: number[];
  pricePerItem: number;
};
