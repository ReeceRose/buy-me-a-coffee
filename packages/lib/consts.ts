export const MAINNET = 'mainnet';
export const TESTNET = 'testnet';
export const DEVNET = 'devnet';
