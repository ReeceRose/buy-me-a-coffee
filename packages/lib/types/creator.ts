export type Creator = {
  username: string;
  description: string;
  bannerImageUrl: string;
  profileImageUrl: string;
};
