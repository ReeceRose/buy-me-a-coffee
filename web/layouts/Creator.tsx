import SEO from '@/components/SEO';
import CreatorNavbar from '@/components/Navbars/CreatorNavbar';

interface Props {
  children: React.ReactNode;
  title: string;
  description: string;
}

export default function CreatorLayout({
  children,
  title,
  description,
}: Props): JSX.Element {
  return (
    <>
      <SEO title={title} description={description} />
      <main>
        <CreatorNavbar />
        {children}
      </main>
    </>
  );
}
