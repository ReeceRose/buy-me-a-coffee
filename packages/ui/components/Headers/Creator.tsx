import Image from 'next/image';

import { CreatorProps } from 'lib/types/props';

export const CreatorHeader = ({ creator }: CreatorProps): JSX.Element => {
  return (
    <>
      <div className="w-full px-6 m-auto text-center">
        <div className="relative h-64 overflow-hidden bg-gray-500 rounded-lg">
          <div className="absolute top-0 right-0 block w-full h-full">
            <Image
              className="object-cover h-full min-w-full"
              layout="fill"
              src={creator.bannerImageUrl}
            />
          </div>
        </div>
      </div>
      <div className="relative flex flex-col w-full mt-16 mb-6 break-words">
        <div className="px-6">
          <div className="flex flex-wrap justify-center">
            <div className="flex justify-center w-full px-4">
              <div className="relative overflow-hidden border-8 border-white rounded-full w-36 h-36 -m-44">
                <Image
                  alt={creator.username}
                  objectFit="cover"
                  layout="fill"
                  src={creator.profileImageUrl}
                  className="absolute align-middle"
                />
              </div>
            </div>
            <div className="w-full px-4 -mt-10 text-center">
              <div className="flex justify-center py-4 pt-8 lg:pt-4">
                <div className="p-3 text-center">
                  <span className="block text-3xl font-bold tracking-wide text-blueGray-600">
                    {creator.username}
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
