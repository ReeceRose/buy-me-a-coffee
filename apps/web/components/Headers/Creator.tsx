import Image from 'next/image';

import { CreatorProps } from '@/lib/props/';

export const CreatorHeader = ({ creator }: CreatorProps): JSX.Element => {
  return (
    <>
      <div className="w-full px-6 m-auto text-center ">
        <div className="relative h-64 overflow-hidden bg-gray-500 rounded-lg">
          <div className="absolute top-0 right-0 block w-full h-full">
            <Image
              className="object-cover h-full min-w-full"
              layout="fill"
              alt="Creator banner image"
              src={creator.bannerImageUrl}
            />
          </div>
        </div>
      </div>
      <div className="relative flex flex-col w-full mt-16 mb-6 break-words">
        <div className="px-6">
          <div className="flex flex-wrap justify-center">
            <div className="flex justify-center w-full px-4">
              <div className="relative overflow-hidden border-8 border-white rounded-full w-36 h-36 -m-44">
                <Image
                  alt={creator.username}
                  objectFit="cover"
                  layout="fill"
                  src={creator.profileImageUrl}
                  className="absolute align-middle bg-gray-500"
                />
              </div>
            </div>
            <div className="w-full px-4 -mt-12 text-center ">
              <div className="flex justify-center pt-8 lg:pt-4">
                <div className="text-center ">
                  <span className="block text-3xl font-bold tracking-wide">
                    {creator.username}
                  </span>
                  <span className="block mt-1 text-xl tracking-wide">
                    {creator.description}
                  </span>
                  <span className="block mt-5 text-lg tracking-wide text-gray-500">
                    {creator.supporters} supporters
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
