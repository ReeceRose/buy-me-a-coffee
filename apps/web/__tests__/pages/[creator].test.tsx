import { GetServerSidePropsContext } from 'next';

import { render } from '@testing-library/react';

import Creator, { getServerSideProps } from '@/pages/[creator]';
import { ParsedUrlQuery } from 'querystring';
import { Creator as CreatorType } from 'lib/types';

describe('Creator page tests', () => {
  const creator: CreatorType = {
    username: 'TestUser',
    description: 'is a test user',
    wallet: 'BiofzXP2MXVFBtMeT5zmoDGkfUTkpHutAbCUkzKW2mxj',
    bannerImageUrl:
      'https://t3.ftcdn.net/jpg/02/68/48/86/360_F_268488616_wcoB2JnGbOD2u3bpn2GPmu0KJQ4Ah66T.jpg',
    profileImageUrl:
      'https://www.pngitem.com/pimgs/m/30-307416_profile-icon-png-image-free-download-searchpng-employee.png',
    supporters: 198,
    support: {
      icon: 'Si/SiBuymeacoffee',
      defaults: [1, 3, 5],
      item: 'coffee',
      pricePerItem: 1.75,
    },
  };

  test('Page matches snapshot', () => {
    const { asFragment } = render(<Creator creator={creator} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('getServerSide props returns expected data', async () => {
    const context = {
      query: { creator: 'TestUser' } as ParsedUrlQuery,
    };

    const response = await getServerSideProps(
      context as GetServerSidePropsContext
    );

    // TODO: mock API to return data
    expect(response).toEqual({
      props: {
        creator: creator,
      },
    });
  });
});
