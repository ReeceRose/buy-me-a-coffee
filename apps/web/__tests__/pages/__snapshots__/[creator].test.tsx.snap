// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator page tests Page matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <nav
      class="fixed top-0 z-50 flex flex-wrap items-center justify-between w-full px-2 py-3 bg-transparent"
    >
      <div
        class="container flex flex-wrap items-center justify-between px-4 mx-auto"
      >
        <div
          class="flex items-center flex-grow shadow-none "
          id="nav"
        >
          <ul
            class="flex flex-row ml-auto list-none"
          >
            <li
              class="flex items-center"
            >
              <a
                class="mr-2 text-lg text-white leading-lg"
                href="/"
              >
                <span
                  class="flex items-center"
                >
                  <svg
                    class="mr-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 13h1v7c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-7h1a1 1 0 0 0 .707-1.707l-9-9a.999.999 0 0 0-1.414 0l-9 9A1 1 0 0 0 3 13zm7 7v-5h4v5h-4zm2-15.586 6 6V15l.001 5H16v-5c0-1.103-.897-2-2-2h-4c-1.103 0-2 .897-2 2v5H6v-9.586l6-6z"
                    />
                  </svg>
                  Home
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="flex items-center content-center justify-center min-w-full"
    >
      <div
        class="container min-w-full mx-auto"
      >
        <div
          class="flex flex-wrap items-center"
        >
          <div
            class="w-full px-6 m-auto text-center"
          >
            <div
              class="relative h-64 overflow-hidden bg-gray-500 rounded-lg"
            >
              <div
                class="absolute top-0 right-0 block w-full h-full"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    class="object-cover h-full min-w-full"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <div
            class="relative flex flex-col w-full mt-16 mb-6 break-words"
          >
            <div
              class="px-6"
            >
              <div
                class="flex flex-wrap justify-center"
              >
                <div
                  class="flex justify-center w-full px-4"
                >
                  <div
                    class="relative overflow-hidden border-8 border-white rounded-full w-36 h-36 -m-44"
                  >
                    <span
                      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                    >
                      <img
                        alt="TestUser"
                        class="absolute align-middle"
                        data-nimg="fill"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                      />
                      <noscript />
                    </span>
                  </div>
                </div>
                <div
                  class="w-full px-4 -mt-10 text-center"
                >
                  <div
                    class="flex justify-center py-4 pt-8 lg:pt-4"
                  >
                    <div
                      class="p-3 text-center"
                    >
                      <span
                        class="block text-3xl font-bold tracking-wide text-blueGray-600"
                      >
                        TestUser
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
